apps/omg/lib/omg/ethereum_event_listener.ex

```diff
@@ -110,13 +111,17 @@ defmodule OMG.EthereumEventListener do
     end
   end
 
-  defp sync_height({%Core{} = state, callbacks}, %SyncGuide{sync_height: sync_height} = sync_info) do
+  defp sync_height(
+         {%Core{service_name: service_name} = state, callbacks},
+         %SyncGuide{sync_height: sync_height} = sync_info
+       ) do
     {:ok, events, db_updates, height_to_check_in, new_state} =
       Core.get_events_range_for_download(state, sync_info)
       |> maybe_update_event_cache(callbacks.get_ethereum_events_callback)
       |> Core.get_events(sync_height)
 
     {:ok, db_updates_from_callback} = callbacks.process_events_callback.(events)
+    Metrics.increment(service_name |> Atom.to_string(), length(events))
```

Comment:
this seems very intrusive and we're mixing decorators and explicit metrics

Reply:
It reminds me more Template method pattern. 
Do you suggest add one more callback?

Reply:
this needs to be done with `:telemetry`.